torch/utils/benchmark/utils/valgrind_wrapper/timer_interface.py

```diff
@@ -124,53 +220,168 @@ def as_standardized(self) -> "CallgrindStats":
-                    fn = re.sub(r"^.+/" + re.escape(new_prefix), new_prefix, fn)
+                # "Python" and "Objects" come from CPython.
+                (r"^.+/" + re.escape("Python/"), "Python/"),
+                (r"^.+/" + re.escape("Objects/"), "Objects/"),
 
                 # Strip library name. e.g. `libtorch.so`
-                fn = re.sub(r"\s\[.+\]$", "", fn)
-                counts[fn] += c
-            return tuple(sorted([
-                FunctionCount(c, fn) for fn, c in counts.items() if c
-            ], reverse=True))
+                (r"\s\[.+\]$", ""),
+            )
+
+            for before, after in transforms:
+                stats = stats.transform(lambda fn: re.sub(before, after, fn))
+
+            return stats
 
         return CallgrindStats(
-            stmt=self.stmt,
-            setup=self.setup,
+            task_spec=self.task_spec,
             number_per_run=self.number_per_run,
-            num_threads=self.num_threads,
             built_with_debug_symbols=self.built_with_debug_symbols,
             baseline_inclusive_stats=strip(self.baseline_inclusive_stats),
             baseline_exclusive_stats=strip(self.baseline_exclusive_stats),
             stmt_inclusive_stats=strip(self.stmt_inclusive_stats),
             stmt_exclusive_stats=strip(self.stmt_exclusive_stats),
         )
 
-    @staticmethod
-    def _counts(stats: Tuple[FunctionCount, ...], include_lookdict_unicode: bool) -> int:
-        return sum(
-            c for c, fn in stats
-            if include_lookdict_unicode
-            or "dictobject.c:lookdict_unicode" not in fn
-        )
 
-    @staticmethod
-    def _diff(first: Tuple[FunctionCount, ...], second: Tuple[FunctionCount, ...]) -> Tuple[FunctionCount, ...]:
-        counts = collections.defaultdict(int, {fn: c for c, fn in first})
-        assert len(counts) == len(first)
-        for c, fn in second:
-            counts[fn] -= c
+class Serialization(enum.Enum):
+    PICKLE = 0
+    TORCH = 1
+    TORCH_JIT = 2
+
+
+_GLOBALS_TYPES_WHITELIST: Dict[Serialization, Tuple[Any, ...]] = {
+    Serialization.PICKLE: (str, bytes, bool, int, float, complex),
+    Serialization.TORCH_JIT: (torch.jit.ScriptFunction, torch.jit.ScriptModule),
+    Serialization.TORCH: (torch.nn.Module,),
+}
+
+
+class GlobalsBridge:
+    """Handle the transfer of (certain) globals when collecting Callgrind statistics.
+
+    Consider the following code snippet:
+    ```
+        import pickle
+        import timeit
+
+        class Counter:
+            value = 0
+
+            def __call__(self):
+                self.value += 1
+
+        counter = Counter()
+        timeit.Timer("counter()", globals={"counter": counter}).timeit(10)
+        print(counter.value)  # 10
+
+        timeit.Timer(
+            "counter()",
+            globals={"counter": pickle.loads(pickle.dumps(counter))}
+        ).timeit(20)
+        print(counter.value)  # Still 10
```

Comment:
I must admit, I'm not terribly convinced by this argument. Yes, side effects run inside the timer may get disregarded. But there isn't really any reason to presuppose that the user cared about the side effects at all in the first place. After all, they're calling a *timer* on a piece of code in a loop. It's incredibly unlikely that they actually wanted to call the operation 10 times; if they are doing a side-effectful operation, they're just doing it to exercise some piece of code that they're interested in timing.

If the user doesn't care about the side effects, then blacklisting Tensor seems like it's just unnecessarily making people's life harder when they have some nontrivial setup that they don't want to have to handle inside Timer.

A *better* reason to block globals is if the serialization/deserialization process perturbs the representation of a tensor in such a way that would result in the timing to be different. This is not an idle concern; for example, if a tensor lives in pinned CPU memory, I'm reasonably certain this wouldn't get preserved by a dump, and that will change the performance of certain CUDA operations.

Reply:
You don't have to do this in this PR, but some more thoughts: it seems better and more explicit to make the user say that they are doing some sort of serialization. Allow the transfer of primitive types (where the serialization is well defined) but then make the user actually do serialization/deserialization if they want to. It will make it more obvious that something is going on (and if there is a bug in the user's serialization code, it will save them a lot of heartbreak).

Reply:
I've updated the language, and added a `CopyIfCallgrind` wrapper for users to declare that they're willing to have their classes serialized. I realized that there's a slight chicken and egg problem. You might want `setup` to execute before globals are loaded so you can setup the environment for unpickle to succeed, but you would want the reverse if you plan to use `setup` to revive bytes in `globals`. `timeit` has the latter behavior (I thought it was the former so I need to switch the codegen order), but that seems to imply that `CopyIfCallgrind` has to also allow an optional per-variable setup.

Reply:
```
"model": benchmark_utils.CopyIfCallgrind(
    MyModule(),
    setup=f"""\
    import sys
    sys.path.append({repr(os.path.split(os.path.abspath(__file__))[0])})
    from test_benchmark_utils import MyModule
    """
)
```

Isn't pickle **fun**?

Reply:
(Also because pickle recursively unpickles I don't think we can automagically generate the `CopyIfCallgrind.setup` code)